pkg/virt-controller/watch/vmi.go

```diff
@@ -2447,3 +2449,51 @@ func (c *VMIController) syncMemoryHotplug(vmi *virtv1.VirtualMachineInstance) {
 		vmi.Labels[virtv1.MemoryHotplugOverheadRatioLabel] = *overheadRatio
 	}
 }
+
+func (c *VMIController) aggregateDataVolumesConditions(vmiCopy *virtv1.VirtualMachineInstance, dvs []*cdiv1.DataVolume) {
+	if len(dvs) == 0 {
+		return
+	}
+
+	dvsReadyCondition := virtv1.VirtualMachineInstanceCondition{
+		Status:  k8sv1.ConditionTrue,
+		Type:    virtv1.VirtualMachineInstanceDataVolumesReady,
+		Reason:  virtv1.VirtualMachineInstanceReasonAllDVsReady,
+		Message: "All of the VMI's DVs are bound and not running",
+	}
+
+	var dvMessages []string
+	for _, dv := range dvs {
+		cStatus, cMessage := getReadyConditionStatus(dv.Status.Conditions)
+		if cStatus != k8sv1.ConditionTrue {
+			dvsReadyCondition.Reason = virtv1.VirtualMachineInstanceReasonNotAllDVsReady
+			if cMessage != "" {
+				dvMessages = append(dvMessages, cMessage)
+			}
+			if cStatus == k8sv1.ConditionFalse {
+				dvsReadyCondition.Status = cStatus
+			} else if cStatus == k8sv1.ConditionUnknown && dvsReadyCondition.Status == k8sv1.ConditionTrue {
```

Comment:
redundant `else`

Reply:
it's the if `cStatus == k8sv1.ConditionUnknown` that is redundant I think

Reply:
@tiraboschi Following your reasoning, it looks like w are not covering the case were there are more than one dv with `Unknown` status: i.e.: DV-1: `True`, DV-2: `Unknown`, DV-3: `Unknown` -> should report `Unknown`.
Also, as @acardace suggested, the `cStatus == k8sv1.ConditionUnknown` part is redundant because:
- we are inside the `cStatus != k8sv1.ConditionTrue`  if;
- we are in the else if of `cStatus == k8sv1.ConditionFalse`.

So summarizing I think this could be:

```suggestion
            // current global ready condition is Unknown or True
			} else if dvsReadyCondition.Status != k8sv1.ConditionFalse {
```

Reply:
done